app/src/main/java/com/guidedmeditationtreks/vipassana/models/TrackTemplate.java

```diff
@@ -0,0 +1,52 @@
+package com.guidedmeditationtreks.vipassana.models;
+
+import android.media.MediaPlayer;
+
+import java.io.File;
+
+/**
+ * Created by aerozero on 12/21/17.
+ */
+
+public class TrackTemplate {
+
+    public String name;
+    public String part1FileName;
+    public MediaPlayer part1MediaPlayer;
+    public int part1Duration;
+
+    public String part2FileName;
+    public MediaPlayer part2MediaPlayer;
+    public int part2Duration;
+
+    public int minimumDuration;
+    public boolean isMultiPart;
+
+    public TrackTemplate(String name, String part1FileName, String part2FileName) {
+
+        this.name = name;
+        this.part1FileName = part1FileName;
+        this.part1MediaPlayer = new MediaPlayer();
+
+        try {
+            this.part1MediaPlayer.setDataSource("/app/src/main/res/audio" + File.separator + part1FileName);
```

Comment:
we are probably expecting an error to occur here because that file path is going to come up wrong. these are some hints we can try: https://stackoverflow.com/questions/7499605/how-to-play-the-audio-files-directly-from-res-raw-folder https://developer.android.com/guide/topics/media/mediaplayer.html - also read up on mediaplayer, we may need to call prepare() in a different thread and only as we need it instead of up front. we will see performance when we run it and can tweak then...

Reply:
Going to wrestle with this even though my trackTemplateFactory.java is listing many errors do an IDE resolution problem